src/shared/sam/debugger/goSamDebug.ts

```diff
@@ -28,8 +33,16 @@ export async function invokeGoLambda(ctx: ExtContext, config: GoDebugConfigurati
     // eslint-disable-next-line @typescript-eslint/unbound-method
     config.onWillAttachDebugger = waitForDelve
 
-    if (!config.noDebug) {
-        await installDebugger(config.debuggerPath!)
+    if (!config.noDebug && !(await installDebugger(config.debuggerPath!))) {
+        showErrorWithLogs(
+            localize(
+                'AWS.sam.debugger.godelve.failed',
+                'Failed to install Delve for the lambda container. See logs for details.'
+            )
+        )
+
+        // Terminates the debug session by sending up a dummy config
+        return {} as GoDebugConfiguration
```

Comment:
This is a nice trick! Wonder if we can use this for the other runtimes.

Reply:
Should be able to. It's important not to send the actual config back, otherwise other extensions could try to run it.